ui/pages/confirmations/hooks/alerts/signatures/useAccountMismatchAlerts.ts

```diff
@@ -0,0 +1,54 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../selectors';
+import { getConfirmationSender } from '../../../components/confirm/utils';
+import { SignatureRequestType } from '../../../types/confirm';
+import { isSIWESignatureRequest } from '../../../utils';
+
+/**
+ * This hook returns an array of alerts when the expected address of the request
+ * does not match the selected account's address.
+ */
+export default function useAccountMismatchAlerts(): Alert[] {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+  const { from: fromAddress } = getConfirmationSender(currentConfirmation);
+
+  const isSIWE = isSIWESignatureRequest(currentConfirmation);
+  const siweParsedAddress =
+    currentConfirmation?.msgParams?.siwe?.parsedMessage?.address;
+  const isMismatchSIWEAdddress =
+    siweParsedAddress?.toLowerCase() !== fromAddress?.toLowerCase();
+  const isMismatchAccount = isSIWE && isMismatchSIWEAdddress;
+
+  const alerts = useMemo(() => {
+    if (!isMismatchAccount) {
+      return [];
+    }
+
+    return [
+      {
+        field: 'account',
+        key: 'account',
+        message: t('alertMessageSignInWrongAccount'),
+        reason: t('alertReasonWrongAccount'),
+        severity: Severity.Warning,
+      },
+      {
+        field: 'signingInWith',
```

Comment:
Why are we adding a single alert to two fields?

I understood our alert system assumes alerts are general or targeted to a single field?

Can we not pick the most relevant single field such as signing in with?

Reply:
oh I think you may be right. I was referring to the design in the screenshot but in Figma, it does indeed look like the alert only exists on "Signing in with". Let me double-check w/ design

Reply:
thanks for comment! at least for this release, we will want to show only 1 alert on "Signing in with". I'll create a new PR for this in case we'd like to show the same alert on two different rows in the future